### PR TITLE
KAFKA-5379 - ProcessorContext.appConfigs() should return parsed values.

### PR DESCRIPTION
**Pull request by twbecker**

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk7-scala2.11/5769/
Test FAILed (JDK 7 and Scala 2.11).

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk8-scala2.12/5755/
Test FAILed (JDK 8 and Scala 2.12).

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk7-scala2.11/5770/
Test PASSed (JDK 7 and Scala 2.11).

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk8-scala2.12/5756/
Test PASSed (JDK 8 and Scala 2.12).

File: streams/src/main/java/org/apache/kafka/streams/processor/ProcessorContext.java
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/processor/ProcessorContext.java]: Nit: `{@link org.apache.kafka.common.config.ConfigDef.Type type}` similar below.

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractProcessorContext.java
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractProcessorContext.java]: nit: add `final`

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/AbstractProcessorContextTest.java
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/AbstractProcessorContextTest.java]: Add a comment that this must be a String type to make a useful test.

**Review by mjsax — CHANGES_REQUESTED** [review]
Can you also fix `RocksDBStore#openDB()` to use this new functionality to avoid the parsing step there. Thx.

File: streams/src/main/java/org/apache/kafka/streams/processor/ProcessorContext.java
Comment (twbecker) [reply streams/src/main/java/org/apache/kafka/streams/processor/ProcessorContext.java]: Done.

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractProcessorContext.java
Comment (twbecker) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractProcessorContext.java]: Done.

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/AbstractProcessorContextTest.java
Comment (twbecker) [reply streams/src/test/java/org/apache/kafka/streams/processor/internals/AbstractProcessorContextTest.java]: Done.

**Comment by twbecker** [comment]
Updated `RocksDBStore` as you suggested. I also updated the test to merely ensure the `RocksDBConfigSetter` is called, since the String vs Class config variations are now tested elsewhere.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk8-scala2.12/5783/
Test PASSed (JDK 8 and Scala 2.12).

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk7-scala2.11/5797/
Test FAILed (JDK 7 and Scala 2.11).

**Review by mjsax — APPROVED** [review]
LGTM.

Call for second review @enothereska @bbejeck

**Comment by mjsax** [comment]
Unclear why test failed. But it was in core, so unrelated.

**Comment by mjsax** [comment]
Retest this please.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk7-scala2.11/5815/
Test PASSed (JDK 7 and Scala 2.11).

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk8-scala2.12/5801/
Test FAILed (JDK 8 and Scala 2.12).

**Comment by mjsax** [comment]
Test failed due to environmental issue...

**Comment by mjsax** [comment]
Retest this please.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk7-scala2.11/5819/
Test PASSed (JDK 7 and Scala 2.11).

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk8-scala2.12/5805/
Test PASSed (JDK 8 and Scala 2.12).

**Comment by mjsax** [comment]
@twbecker There are checkstyle errors in the build. Will review after those got fixed. Thx.

**Comment by twbecker** [comment]
Yeah, that's what I get for fixing the conflicts in the Github web UI ;)

**Comment by guozhangwang** [comment]
retest this please

**Comment by guozhangwang** [comment]
retest this please

**Comment by guozhangwang** [comment]
retest this please

**Comment by guozhangwang** [comment]
@twbecker The jenkins failures are related:

```
:streams:checkstyleMain[ant:checkstyle] [ERROR] /home/jenkins/jenkins-slave/workspace/kafka-pr-jdk7-scala2.11@2/streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java:20:8: Unused import - org.apache.kafka.common.config.ConfigDef. [UnusedImports]
 FAILED
```

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBStoreTest.java
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBStoreTest.java]: Nit: add `final` to both parameters -- please follow a "one parameter per line" formatting.

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBStoreTest.java
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBStoreTest.java]: Nit: `final`

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBStoreTest.java
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBStoreTest.java]: Nit: `final` and formatting

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBStoreTest.java
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBStoreTest.java:334]: Why not stay with `MockProcessorContext` -- this should avoid to overwrite all the empty methods below?

**Review by mjsax — CHANGES_REQUESTED** [review]
Thanks for the PR @twbecker !

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBStoreTest.java
Comment (twbecker) [reply streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBStoreTest.java:334]: It's been a while since I did this, but I think it was because I needed to specify the `StreamsConfig`.

**Comment by twbecker** [comment]
Ok, @mjsax should be gtg now.

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBStoreTest.java
Comment (guozhangwang) [reply streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBStoreTest.java:334]: I think we can do it in a follow-up PR after merging this one.

**Comment by guozhangwang** [comment]
Merged to trunk. Thanks @twbecker .

**Comment by mjsax** [comment]
Thanks @twbecker !

**Pull request closed without merging**
